### PR TITLE
Random cleanup

### PR DESCRIPTION
**Pull request by kamilritz**
This PR contains three cleanups:

1) Some variables containing the acronym *mag*  for magnitude are renamed to avoid confusion.
On top, some class variables are relocated.

2) Once we acquire a Gps location we can compute the earth rotation vector in our local coordinate system. This was done in the `predictState()` function. **Misplaced responsibility**. Porting this to gps_checks.cpp save one variable.

3) While on ground we check GPS signal for drift. Previously we had a variable for each component of the position derivative. This PR adds vector quantities to reduce number of variables, lines of code and increase the readability of the code.

SITL mission test: [Log](https://review.px4.io/plot_app?log=55106fda-c011-48ee-a8cf-7958e853fa64)

### CODE REVIEW

File: EKF/gps_checks.cpp [hunk 1]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
-		float velE = fminf(fmaxf(delta_PosE / dt, -vel_limit), vel_limit);
+		// Calculate the horizontal and vertiacl drift velocity components and limit to 10x the threshold
+		const Vector3f vel_limit{_params.req_hdrift, _params.req_hdrift, _params.req_vdrift};
+		Vector3f pos_derived{delta_posN, delta_posE, (_gps_alt_prev - 1e-3f * (float)gps.alt)};
+		pos_derived = matrix::constrain(pos_derived / dt, -10.0f * vel_limit, 10.0f * vel_limit);
 
 		// Apply a low pass filter
-		_gpsDriftVelN = velN * filter_coef + _gpsDriftVelN * (1.0f - filter_coef);
-		_gpsDriftVelE = velE * filter_coef + _gpsDriftVelE * (1.0f - filter_coef);
+		_gps_pos_derived = pos_derived * filter_coef + _gps_pos_derived * (1.0f-filter_coef);
```
Comment (bresch) [inline EKF/gps_checks.cpp]:
```suggestion
		_gps_pos_derived_filt = pos_derived * filter_coef + _gps_pos_derived * (1.0f-filter_coef);
```

File: EKF/ekf.h [hunk 2]
```diff
@@ -465,15 +462,13 @@ class Ekf : public EstimatorInterface
...
 	uint32_t _mag_counter{0};		///< number of magnetometer samples read during initialisation
-	uint64_t _time_last_mag{0};		///< measurement time of last magnetomter sample (uSec)
 	AlphaFilterVector3f _mag_lpf;		///< filtered magnetometer measurement for instant reset(Gauss)
 	AlphaFilterVector3f _accel_lpf;		///< filtered accelerometer measurement for instant reset(Gauss)
 	float _hgt_sensor_offset{0.0f};		///< set as necessary if desired to maintain the same height after a height reset (m)
 	float _baro_hgt_offset{0.0f};		///< baro height reading at the local NED origin (m)
 
 	// Variables used to control activation of post takeoff functionality
 	float _last_on_ground_posD{0.0f};	///< last vertical position when the in_air status was false (m)
-	bool _flt_mag_align_converging{false};	///< true when the in-flight mag field post alignment convergence is being performd
```
Comment (jkflying) [inline EKF/ekf.h]: I wonder if the place where this was used before was accidentally removed? Could you do a quick search through the git history to make sure this wasn't a mistake?

Comment (kamilritz) [reply EKF/ekf.h]: You are right! We lost the following assignment during the [mag control refactor](https://github.com/PX4/ecl/pull/662/files#diff-4cac31c333867233bb8a08de1e812a9aL1461).
FYI @bresch @priseborough

Comment (bresch) [reply EKF/ekf.h]: Thanks for pointing that, it's going to be fixed in https://github.com/PX4/ecl/pull/692

**Comment by kamilritz** [comment]
@bresch I removed the parts regarding the mag

File: EKF/gps_checks.cpp [hunk 3]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
 			map_projection_init_timestamped(&_gps_pos_prev, lat, lon, _time_last_imu);
 			_gps_alt_prev = 1e-3f * (float)gps.alt;
-
 		}
 
-		// Calculate the horizontal drift velocity components and limit to 10x the threshold
-		float vel_limit = 10.0f * _params.req_hdrift;
-		float velN = fminf(fmaxf(delta_posN / dt, -vel_limit), vel_limit);
-		float velE = fminf(fmaxf(delta_PosE / dt, -vel_limit), vel_limit);
+		// Calculate the horizontal and vertiacl drift velocity components and limit to 10x the threshold
```
Comment (priseborough) [inline EKF/gps_checks.cpp]: "vertiacl"

**Comment by priseborough** [comment]
Can you please rebase. TXS

File: EKF/gps_checks.cpp (hunk 3 above)
Comment (kamilritz) [reply EKF/gps_checks.cpp]: done

**Comment by kamilritz** [comment]
@priseborough rebased on master

File: EKF/gps_checks.cpp [hunk 4]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
 	if (!_control_status.flags.in_air && _vehicle_at_rest) {
 		// Calculate position movement since last measurement
 		float delta_posN = 0.0f;
-		float delta_PosE = 0.0f;
+		float delta_posE = 0.0f;
```
Comment (bresch) [inline EKF/gps_checks.cpp]: would be even better to name it `delta_pos_e`

File: EKF/gps_checks.cpp [hunk 5]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
 			_gps_alt_prev = 1e-3f * (float)gps.alt;
-
 		}
 
-		// Calculate the horizontal drift velocity components and limit to 10x the threshold
-		float vel_limit = 10.0f * _params.req_hdrift;
-		float velN = fminf(fmaxf(delta_posN / dt, -vel_limit), vel_limit);
-		float velE = fminf(fmaxf(delta_PosE / dt, -vel_limit), vel_limit);
+		// Calculate the horizontal and vertical drift velocity components and limit to 10x the threshold
+		const Vector3f vel_limit{_params.req_hdrift, _params.req_hdrift, _params.req_vdrift};
```
Comment (bresch) [inline EKF/gps_checks.cpp]:
```suggestion
		const Vector3f vel_limit(_params.req_hdrift, _params.req_hdrift, _params.req_vdrift);
```

File: EKF/gps_checks.cpp [hunk 6]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
+		Vector3f pos_derived{delta_posN, delta_posE, (_gps_alt_prev - 1e-3f * (float)gps.alt)};
```
Comment (bresch) [inline EKF/gps_checks.cpp]:
```suggestion
		Vector3f pos_derived(delta_posN, delta_posE, (_gps_alt_prev - 1e-3f * (float)gps.alt));
```

File: EKF/gps_checks.cpp [hunk 7]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
+		pos_derived = matrix::constrain(pos_derived / dt, -10.0f * vel_limit, 10.0f * vel_limit);
 
 		// Apply a low pass filter
-		_gpsDriftVelN = velN * filter_coef + _gpsDriftVelN * (1.0f - filter_coef);
-		_gpsDriftVelE = velE * filter_coef + _gpsDriftVelE * (1.0f - filter_coef);
+		_gps_pos_derived_filt = pos_derived * filter_coef + _gps_pos_derived_filt * (1.0f-filter_coef);
```
Comment (bresch) [inline EKF/gps_checks.cpp]:
```suggestion
		_gps_pos_derived_filt = pos_derived * filter_coef + _gps_pos_derived_filt * (1.0f - filter_coef);
```

File: EKF/gps_checks.cpp [hunk 8]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
 
 		// Calculate the horizontal drift speed and fail if too high
-		_gps_drift_metrics[0] = sqrtf(_gpsDriftVelN * _gpsDriftVelN + _gpsDriftVelE * _gpsDriftVelE);
+		_gps_drift_metrics[0] = Vector2f{_gps_pos_derived_filt.xy()}.norm();
```
Comment (bresch) [inline EKF/gps_checks.cpp]:
```suggestion
		_gps_drift_metrics[0] = Vector2f(_gps_pos_derived_filt.xy()).norm();
```

File: EKF/gps_checks.cpp [hunk 9]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
...
 		_gps_check_fail_status.flags.vdrift = (_gps_drift_metrics[1] > _params.req_vdrift);
 
 		// Check the magnitude of the filtered horizontal GPS velocity
-		float vxy_drift_limit = 10.0f * _params.req_hdrift;
-		float gps_velN = fminf(fmaxf(gps.vel_ned[0], -vxy_drift_limit), vxy_drift_limit);
-		float gps_velE = fminf(fmaxf(gps.vel_ned[1], -vxy_drift_limit), vxy_drift_limit);
-		_gps_velN_filt = gps_velN * filter_coef + _gps_velN_filt * (1.0f - filter_coef);
-		_gps_velE_filt  = gps_velE * filter_coef + _gps_velE_filt  * (1.0f - filter_coef);
-		_gps_drift_metrics[2] = sqrtf(_gps_velN_filt * _gps_velN_filt + _gps_velE_filt * _gps_velE_filt);
+		Vector2f gps_velNE = matrix::constrain(Vector2f{gps.vel_ned[0], gps.vel_ned[1]}, // TODO: when vel_ned vector3f use .xy()
```
Comment (bresch) [inline EKF/gps_checks.cpp]:
```suggestion
		Vector2f gps_velNE = matrix::constrain(Vector2f(gps.vel_ned[0], gps.vel_ned[1]), // TODO: when vel_ned vector3f use .xy()
```

**Comment by kamilritz** [comment]
Updated

File: EKF/gps_checks.cpp [hunk 10]
```diff
@@ -145,51 +146,40 @@ bool Ekf::gps_is_good(const gps_message &gps)
 	if (!_control_status.flags.in_air && _vehicle_at_rest) {
 		// Calculate position movement since last measurement
 		float delta_posN = 0.0f;
-		float delta_PosE = 0.0f;
+		float delta_pos_e = 0.0f;
```
Comment (mhkabir) [inline EKF/gps_checks.cpp]: Change delta_posN too if you're doing this one.

File: EKF/ekf.cpp [hunk 11]
```diff
@@ -192,7 +192,7 @@ bool Ekf::initialiseFilter()
 		setControlBaroHeight();
 
 		// reset variables that are shared with post alignment GPS checks
-		_gps_drift_velD = 0.0f;
+		_gps_pos_derived_filt(2) = 0.0f;
```
[the remark this answers is not in the document]
Comment (priseborough) [reply EKF/ekf.cpp]: The quantity being filtered is a position derivative, so yes.

**Pull request merged**